src/include/buffer.h

```diff
@@ -881,6 +881,13 @@ namespace buffer CEPH_BUFFER_API {
     }
     uint32_t crc32c(uint32_t crc) const;
     void invalidate_crc();
+
+    // These functions return a bufferlist with a pointer to a single
+    // static buffer. They /must/ not outlive the memory they
+    // reference.
+    static list static_from_chararray(char* c, size_t l);
+    static list static_from_cstring(char* c);
+    static list static_from_string(std::string& s);
```

Comment:
Ah, I see. As the dependency-minimisation seems to be a key requirement, leaving those methods as-is is perfectly fine.